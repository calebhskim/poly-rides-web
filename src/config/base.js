export default {
  firebaseApp: 'https://polyridesweb.firebaseio.com/',
  groupId: '250502971675365',
  port: 3000,
  urls: {
    graph: 'https://graph.facebook.com/',
  },
};
