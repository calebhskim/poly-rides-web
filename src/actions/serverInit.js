import actions from '../constants/actions';

const serverInit = clientConfig => dispatch => dispatch({
  type: actions.APP_INITIALIZED_SUCCESS,
  payload: {
    config: clientConfig,
  },
});

export default serverInit;
