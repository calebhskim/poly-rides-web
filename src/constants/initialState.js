import lifecycles from './lifecycles';

export default {
  auth: {
    error: {},
    lifecycle: lifecycles.AUTH_NOT_LOGGEDIN,
    user: {
      displayName: null,
      email: null,
      emailVerified: false,
      isAnonymous: null,
      photoURL: null,
      refreshToken: null,
      uid: null,
    },
    fbToken: null,
  },
  config: {
    title: 'PolyRides',
  },
  firebase: {
    app: null,
    fbProvider: null,
  },
};
