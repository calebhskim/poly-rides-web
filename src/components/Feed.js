import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';

import Paper from 'material-ui/Paper';

import feedMedia from '../styles/css/feed.css';
import FeedScroll from './FeedScroll';
import feedStyle from '../styles/components/feed';
import { listenForRides, stopListenForRides } from '../actions/rides';
import setNavTitle from '../actions/setNavTitle';

const {
  feedContainer,
} = feedStyle;

export class Feed extends Component {
  constructor(props) {
    super(props);
    this.handleUpdateInput = this.handleUpdateInput.bind(this);
    this.state = {
      start: '',
    };
  }

  componentWillMount() {
    this.props.listenForRides();
  }

  componentDidMount() {
    this.props.setNavTitle('Feed');
  }

  componentWillUnmount() {
    this.props.stopListenForRides();
  }

  handleUpdateInput(value) {
    this.setState({
      start: value,
    });
  }

  render() {
    return (
      <Paper className={feedMedia.feedFull} style={feedContainer} id='feed'>
        <FeedScroll />
      </Paper>
    );
  }
}


Feed.propTypes = {
  listenForRides: PropTypes.func,
  setNavTitle: PropTypes.func,
  stopListenForRides: PropTypes.func,
};

function mapStateToProps() {
  return {};
}

const mapDispatchToProps = {
  listenForRides,
  setNavTitle,
  stopListenForRides,
};

export default connect(mapStateToProps, mapDispatchToProps)(Feed);
