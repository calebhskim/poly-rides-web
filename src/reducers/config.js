import initialState from '../constants/initialState';

export default configReducer(state = initialState.config, { type, payload }) {
  switch(type) {
    default:
      return state;
  }
}
