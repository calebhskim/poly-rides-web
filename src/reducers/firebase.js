import initialState from '../constants/initialState';
import actions from '../constants/actions';

const firebase = (state = initialState.firebase, { payload, type }) => {
  switch(type) {
    case actions.APP_INITIALIZED_SUCCESS: {
      const { fbApp, fbProvider } = payload;
      return Object.assign({}, state, {
        app: fbApp,
        fbProvider,
      });
    }
    default:
      return state;
  }
};

export default firebase;
