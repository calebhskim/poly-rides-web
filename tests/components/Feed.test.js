import React from 'react';

import cloneDeep from 'lodash/cloneDeep';
import firebase from 'firebase';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import config from '../../src/config/development';
import Feed from '../../src/components/Feed';
import initialState from '../../src/constants/initialState';
import serverInit from '../../src/actions/serverInit';
import Store from '../../src/store';

// connecting to firebase dev
const fbRef = firebase.initializeApp(config.firebase);
const ref = fbRef.database().ref('ridesTest');

describe('<Feed />', () => {
  it('renders properly', () => {
    const modifiedState = cloneDeep(initialState);
    modifiedState.config.ridesDBName = 'ridesTest';
    const store = new Store(modifiedState);

    store.dispatch(serverInit(fbRef));

    const wrapper = shallow(<Feed store={store} />).shallow();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  // it('loads feed correctly from store', () => {
  //   const modifiedState = cloneDeep(initialState);
  //   modifiedState.firebase.app = ref;
  //   modifiedState.config.ridesDBName = 'ridesTest';
  //   const store = new Store(modifiedState);

  //   const wrapper = shallow(<Feed store={store} />).shallow();

  //   expect(shallowToJson(wrapper)).toMatchSnapshot();
  // });
});
