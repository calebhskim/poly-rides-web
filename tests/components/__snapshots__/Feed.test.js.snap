// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feed /> renders properly 1`] = `
<Paper
  circle={false}
  id="about"
  rounded={true}
  style={
    Object {
      "display": "flex",
      "height": "100vh",
    }
  }
  transitionEnabled={true}
  zDepth={1}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      }
    }
  >
    <h1>
      Feed Header
    </h1>
  </View>
</Paper>
`;
